### PR TITLE
record response code in ICDS audit data

### PR DESCRIPTION
**Pull request by snopoke**
##### SUMMARY
record response code in ICDS audit data

Note: in order to add a 'fast' test I moved the agg tests to a sub-package since they use package level setup.

### CODE REVIEW

**Review by emord — APPROVED** [review]
> moved the agg tests to a sub-package since they use package level setup.

Nice. I really like that. it means if tests for new reports are added in the future, they can use their own data instead of relying on the hack-y agg setup

**Comment by snopoke** [comment]
Tests are failing because package level setup in `custom/icds_reports/tests/agg_tests/__init__.py` is being run with the `ErrorOnDbAccessContext`. @millerdev any pointers to fix this? It used to work before moving the tests to a sub-package.

File: custom/icds_reports/tests/test_audit_data.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        if expected_event:
+            self.assertIsNotNone(event)
+            self.assertEqual(event.url, path)
+            self.assertEqual(event.response_code, response_code)
+        else:
+            self.assertIsNone(event)
+
+
+@generate_cases([
+    (DASHBOARD_DOMAIN, '/a/{domain}/login/', 200, True),
```
Comment (calellowitz) [inline custom/icds_reports/tests/test_audit_data.py]: Is this test case correct. I would have thought the icds cas login page did not trigger the dashboard audit code

**Review by calellowitz — APPROVED** [review]
Assuming I am wrong about how audit code works, this looks good

File: custom/icds_reports/tests/test_audit_data.py (hunk 1 above)
Comment (snopoke) [reply custom/icds_reports/tests/test_audit_data.py]: It does: https://github.com/dimagi/commcare-hq/blob/cc93bc6ab718bde08e8d16e374d5cc25eff94958/custom/icds_reports/middleware.py#L59

**Comment by millerdev** [comment]
> Tests are failing because package level setup ... any pointers to fix this?

Did you intend to remove `__init__.py` from [custom/icds_reports/tests](https://github.com/dimagi/commcare-hq/tree/f2f1133ee5f57c2fbb14f2f2fa3964084027550f/custom/icds_reports/tests)? I think it is now a [namespace package](https://packaging.python.org/guides/packaging-namespace-packages/). I'm guessing you did not intend for it to be a namespace package, so adding an empty `__init__.py` in there may fix it.

If that does not fix it then I'm guessing it has something to do with the way django-nose is differentiating between database and non-database tests when package-level setup is involved. Let me know if you want me to look into that.

**Comment by snopoke** [comment]
> Did you intend to remove __init__.py 

oops, let me add that back.

**Comment by snopoke** [comment]
Build is passing now. Thanks @millerdev

**Pull request merged**